src/components/MasterModal.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { Button, Icon, Modal, Message } from "semantic-ui-react";
+import RegistrationForm from "./RegistrationForm";
+import { performAuthentication } from "../modules/auth";
+import { useDispatch, useSelector } from "react-redux";
+
+const MasterModal = () => {
+  const dispatch = useDispatch();
+  const { errorMessage, authenticated, currentUser } = useSelector(
+    (state) => state
+  );
+  const [firstOpen, setFirstOpen] = useState(false);
+  const [secondOpen, setSecondOpen] = useState(false);
+
+  const registerAndProceed = (e) => {
+    e.preventDefault();
+    performAuthentication(e, dispatch);
+    setSecondOpen(authenticated);
+  };
+
+  const finalizePayment = () => {
+    // Stripe payment functionality will be added here
+    setSecondOpen(false);
+    setFirstOpen(false);
+  };
+
+  return (
+    <>
+      <Button onClick={() => setFirstOpen(true)} data-cy="register-btn">
+        Register To View Premium Content
+      </Button>
+      <Modal
+        onClose={() => setFirstOpen(false)}
+        onOpen={() => setFirstOpen(true)}
+        open={firstOpen}
+        data-cy="first-registration"
+      >
+        <Modal.Header>Enter Your Details</Modal.Header>
+        <Modal.Content>
+          {/* <RegistrationForm /> */}
```

Comment:
This is commented out?

Reply:
Removed the commented out code in both instances